### ISSUE TITLE
Stage 0 fails ("repo" and "common packages")

### ISSUE THREAD

**Issue by smithfarm** [SUSE/DeepSea#1107]
### Description of Issue/Question

In a wip branch based on master 543b06a754a1f2832b552685af0dbf591aabe133 Stage 0 fails when run via the health-ok.sh integration test script:

```
2018-04-23T16:08:11.682 INFO:teuthology.orchestra.run.target147135134182.stdout:No minions matched the target. No command was sent, no jid was assigned.
2018-04-23T16:08:11.682 INFO:teuthology.orchestra.run.target147135134182.stdout:deepsea_minions          : [1m[92mvalid[0m
2018-04-23T16:08:11.683 INFO:teuthology.orchestra.run.target147135134182.stdout:master_minion            : [1m[92mvalid[0m
2018-04-23T16:08:11.683 INFO:teuthology.orchestra.run.target147135134182.stdout:ceph_version             : [1m[92mvalid[0m
2018-04-23T16:08:11.683 INFO:teuthology.orchestra.run.target147135134182.stdout:salt_version             : [1m[92mvalid[0m
2018-04-23T16:08:11.683 INFO:teuthology.orchestra.run.target147135134182.stdout:No minions matched the target. No command was sent, no jid was assigned.
2018-04-23T16:08:11.683 INFO:teuthology.orchestra.run.target147135134182.stdout:No minions matched the target. No command was sent, no jid was assigned.
2018-04-23T16:08:11.684 INFO:teuthology.orchestra.run.target147135134182.stdout:target147135134182.teuthology_master:
2018-04-23T16:08:11.684 INFO:teuthology.orchestra.run.target147135134182.stdout:  Name: sync master - Function: salt.state - Result: Changed Started: - 16:03:50.619388 Duration: 1465.724 ms
2018-04-23T16:08:11.684 INFO:teuthology.orchestra.run.target147135134182.stdout:  Name: salt-api - Function: salt.state - Result: Changed Started: - 16:03:52.085354 Duration: 11370.346 ms
2018-04-23T16:08:11.684 INFO:teuthology.orchestra.run.target147135134182.stdout:  Name: repo master - Function: salt.state - Result: Clean Started: - 16:04:03.456110 Duration: 461.272 ms
2018-04-23T16:08:11.684 INFO:teuthology.orchestra.run.target147135134182.stdout:  Name: prepare master - Function: salt.state - Result: Changed Started: - 16:04:03.917698 Duration: 238137.924 ms
2018-04-23T16:08:11.685 INFO:teuthology.orchestra.run.target147135134182.stdout:  Name: filequeue.remove - Function: salt.runner - Result: Changed Started: - 16:08:02.056625 Duration: 7057.825 ms
2018-04-23T16:08:11.685 INFO:teuthology.orchestra.run.target147135134182.stdout:  Name: filequeue.enqueue - Function: salt.runner - Result: Changed Started: - 16:08:09.114809 Duration: 1004.832 ms
2018-04-23T16:08:11.685 INFO:teuthology.orchestra.run.target147135134182.stdout:  Name: minions.ready - Function: salt.runner - Result: Changed Started: - 16:08:10.120065 Duration: 1393.551 ms
2018-04-23T16:08:11.685 INFO:teuthology.orchestra.run.target147135134182.stdout:----------
2018-04-23T16:08:11.685 INFO:teuthology.orchestra.run.target147135134182.stdout:          ID: repo
2018-04-23T16:08:11.685 INFO:teuthology.orchestra.run.target147135134182.stdout:    Function: salt.state
2018-04-23T16:08:11.686 INFO:teuthology.orchestra.run.target147135134182.stdout:      Result: False
2018-04-23T16:08:11.686 INFO:teuthology.orchestra.run.target147135134182.stdout:     Comment: No minions returned
2018-04-23T16:08:11.686 INFO:teuthology.orchestra.run.target147135134182.stdout:     Started: 16:08:11.514136
2018-04-23T16:08:11.686 INFO:teuthology.orchestra.run.target147135134182.stdout:    Duration: 68.64 ms
2018-04-23T16:08:11.686 INFO:teuthology.orchestra.run.target147135134182.stdout:     Changes:
2018-04-23T16:08:11.687 INFO:teuthology.orchestra.run.target147135134182.stdout:----------
2018-04-23T16:08:11.687 INFO:teuthology.orchestra.run.target147135134182.stdout:          ID: common packages
2018-04-23T16:08:11.687 INFO:teuthology.orchestra.run.target147135134182.stdout:    Function: salt.state
2018-04-23T16:08:11.687 INFO:teuthology.orchestra.run.target147135134182.stdout:      Result: False
2018-04-23T16:08:11.687 INFO:teuthology.orchestra.run.target147135134182.stdout:     Comment: No minions returned
2018-04-23T16:08:11.687 INFO:teuthology.orchestra.run.target147135134182.stdout:     Started: 16:08:11.583222
2018-04-23T16:08:11.688 INFO:teuthology.orchestra.run.target147135134182.stdout:    Duration: 67.53 ms
2018-04-23T16:08:11.688 INFO:teuthology.orchestra.run.target147135134182.stdout:     Changes:
2018-04-23T16:08:11.688 INFO:teuthology.orchestra.run.target147135134182.stdout:
2018-04-23T16:08:11.688 INFO:teuthology.orchestra.run.target147135134182.stdout:Summary for target147135134182.teuthology_master
2018-04-23T16:08:11.688 INFO:teuthology.orchestra.run.target147135134182.stdout:------------
2018-04-23T16:08:11.688 INFO:teuthology.orchestra.run.target147135134182.stdout:Succeeded: 7 (changed=6)
2018-04-23T16:08:11.689 INFO:teuthology.orchestra.run.target147135134182.stdout:Failed:    2
2018-04-23T16:08:11.689 INFO:teuthology.orchestra.run.target147135134182.stdout:------------
2018-04-23T16:08:11.689 INFO:teuthology.orchestra.run.target147135134182.stdout:Total states run:     9
2018-04-23T16:08:11.689 INFO:teuthology.orchestra.run.target147135134182.stdout:Total run time: 261.028 s
```

journalctl -r has the following hint:

```
2018-04-23T16:08:11.896 INFO:teuthology.orchestra.run.target147135134182.stdout:Apr 23 16:08:11 target147135134182.teuthology salt-master[6862]: [DEBUG   ] Gathering reactors for tag salt/run/20180423160321595308/ret
2018-04-23T16:08:11.897 INFO:teuthology.orchestra.run.target147135134182.stdout:Apr 23 16:08:11 target147135134182.teuthology salt-master[6862]: [ERROR   ] Invalid compound target:
2018-04-23T16:08:11.897 INFO:teuthology.orchestra.run.target147135134182.stdout:Apr 23 16:08:11 target147135134182.teuthology salt-master[6862]: [DEBUG   ] Evaluating final compound matching expr:
2018-04-23T16:08:11.897 INFO:teuthology.orchestra.run.target147135134182.stdout:Apr 23 16:08:11 target147135134182.teuthology salt-master[6862]: [DEBUG   ] minions: {'target147135134182.teuthology'}
2018-04-23T16:08:11.897 INFO:teuthology.orchestra.run.target147135134182.stdout:Apr 23 16:08:11 target147135134182.teuthology salt-master[6862]: [DEBUG   ] Gathering reactors for tag salt/run/20180423160321595308/args
2018-04-23T16:08:11.897 INFO:teuthology.orchestra.run.target147135134182.stdout:Apr 23 16:08:11 target147135134182.teuthology salt-master[6862]: [ERROR   ] Invalid compound target:
2018-04-23T16:08:11.898 INFO:teuthology.orchestra.run.target147135134182.stdout:Apr 23 16:08:11 target147135134182.teuthology salt-master[6862]: [DEBUG   ] Evaluating final compound matching expr:
2018-04-23T16:08:11.898 INFO:teuthology.orchestra.run.target147135134182.stdout:Apr 23 16:08:11 target147135134182.teuthology salt-master[6862]: [DEBUG   ] minions: {'target147135134182.teuthology'}
```

The "invalid compound target" error seems suspicious.

I'm now trying if the problem is reproducible in a manual deployment.

**Comment by smithfarm** [comment]
The issue is reproducible in vagrant as well.

```
localhost:/home/vagrant/DeepSea # salt-run state.orch ceph.stage.0       
[WARNING ] /usr/lib/python3.6/site-packages/salt/client/__init__.py:711: DeprecationWarning: The target type should be passed using the 'tgt_type' argument instead of 'expr_form'. Support for using 'expr_form' will be removed in Salt Fluorine.

No minions matched the target. No command was sent, no jid was assigned.
deepsea_minions          : valid
master_minion            : valid
ceph_version             : valid
salt_version             : valid
[WARNING ] All minions are ready
No minions matched the target. No command was sent, no jid was assigned.
[ERROR   ] No minions returned
No minions matched the target. No command was sent, no jid was assigned.
[ERROR   ] No minions returned
admin.ceph_master:
  Name: sync master - Function: salt.state - Result: Changed Started: - 15:34:41.123104 Duration: 1062.275 ms
  Name: salt-api - Function: salt.state - Result: Changed Started: - 15:34:42.185501 Duration: 12867.943 ms
  Name: repo master - Function: salt.state - Result: Clean Started: - 15:34:55.053562 Duration: 420.036 ms
  Name: prepare master - Function: salt.state - Result: Changed Started: - 15:34:55.473720 Duration: 5638.338 ms
  Name: filequeue.remove - Function: salt.runner - Result: Changed Started: - 15:35:01.112270 Duration: 6258.607 ms
  Name: filequeue.enqueue - Function: salt.runner - Result: Changed Started: - 15:35:07.371008 Duration: 1229.475 ms
  Name: minions.ready - Function: salt.runner - Result: Changed Started: - 15:35:08.600622 Duration: 1634.935 ms
----------
          ID: repo
    Function: salt.state
      Result: False
     Comment: No minions returned
     Started: 15:35:10.235700
    Duration: 106.327 ms
     Changes:   
----------
          ID: common packages
    Function: salt.state
      Result: False
     Comment: No minions returned
     Started: 15:35:10.342154
    Duration: 101.801 ms
     Changes:   

Summary for admin.ceph_master
------------
Succeeded: 7 (changed=6)
Failed:    2
------------
Total states run:     9
Total run time:  29.320 s
```

**Comment by smithfarm** [comment]
Fixed by #1108
